### PR TITLE
refactor: client charge screen migration to compose

### PR DESCRIPTION
**Pull request by AvneetSingh2001**
Jira Task [MM-5](https://mifosforge.jira.com/browse/MM-5)

[MM-5]: https://mifosforge.jira.com/browse/MM-5?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

| XML Screen (before)                                                                                               |  Compose Screen (Now)                                                                                            | 
|-------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
| Screenshot 2024-06-14 at 10 55 06 PM | Screenshot 2024-06-14 at 10 57 14 PM |

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeFragment.kt [hunk 1]
```diff
@@ -17,23 +16,18 @@ import kotlinx.coroutines.launch
...
-import org.mifos.mobile.viewModels.ClientChargeViewModel
 
-/**
- * @author Vishwajeet
- * @since 17/8/16.
- */
+/*
+
 @AndroidEntryPoint
 class ClientChargeFragment : BaseFragment() {
```
Comment (AdityaKumdale) [inline app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeFragment.kt:32]: Is this to be deleted

File: app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeComposeFragment.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package org.mifos.mobile.ui.client_charge
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
```
Comment (AdityaKumdale) [inline app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeComposeFragment.kt]: Unused imports

**Review by AdityaKumdale — CHANGES_REQUESTED** [review]

**Comment by AdityaKumdale** [comment]
add screnshots of previews before-after as well , Rest of the logic is fine

File: app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            }.collect {
+                _clientChargeUiState.value = ClientChargeUiState.Success(it.pageItems.filterNotNull())
+            }
+        }
+    }
+}
+
+sealed class ClientChargeUiState {
+    data object Loading : ClientChargeUiState()
+    data class Error(val message: String?) : ClientChargeUiState()
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeViewModel.kt:95]: necessary to make it nullable ?

File: app/src/main/java/org/mifos/mobile/ui/home/HomeOldFragment.kt [hunk 4]
```diff
@@ -128,6 +127,7 @@ class HomeOldFragment : BaseFragment(), OnRefreshListener {
         super.onResume()
         registerReceiver()
         activity?.invalidateOptionsMenu()
+        (activity as? BaseActivity)?.showToolbar()
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/home/HomeOldFragment.kt:129]: wont MFScaffold take care of this ?

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeFragment.kt (hunk 1 above)
Comment (AvneetSingh2001) [reply app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeFragment.kt:32]: Yes, as Rajan bhaiya told to keep the fragment as well, so kept it as commented code here.

File: app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeViewModel.kt (hunk 3 above)
Comment (AvneetSingh2001) [reply app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeViewModel.kt:95]: yes, as the throwable of catch has nullable message in the internal implementation itself. 
I aim to handle this on the compose screen itself, if the message got is null, the with stringResource we can give our default error.

File: app/src/main/java/org/mifos/mobile/ui/home/HomeOldFragment.kt (hunk 4 above)
Comment (AvneetSingh2001) [reply app/src/main/java/org/mifos/mobile/ui/home/HomeOldFragment.kt:129]: We dont have Scaffold implemented on Home Screen as of now, so have to explicitly tell to show toolbar.

File: app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeComposeFragment.kt (hunk 2 above)
Comment (AvneetSingh2001) [reply app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeComposeFragment.kt]: updated. please resolve it.

File: app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeViewModel.kt (hunk 3 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/client_charge/ClientChargeViewModel.kt:95]: understood

**Pull request merged**